ts/components/messages/paginated/MessageDetail/common/HeaderDueDateBar.tsx

```diff
@@ -21,28 +17,34 @@ type Props = {
  * @param props
  * @constructor
  */
-export const HeaderDueDateBar = (props: Props): React.ReactElement => (
-  <>
-    {props.prescriptionData && (
-      <MedicalPrescriptionIdentifiersComponent
-        prescriptionData={props.prescriptionData}
-      />
-    )}
+export const HeaderDueDateBar = (props: Props): React.ReactElement => {
+  const paymentExpirationInfo = getPaymentExpirationInfo(props.messageDetails);
 
-    {props.prescriptionData === undefined && props.dueDate !== undefined && (
-      <DueDateBar
-        dueDate={props.dueDate}
-        expirationInfo={props.paymentExpirationInfo}
-        isPaid={props.hasPaidBadge}
-      />
-    )}
+  return (
+    <>
+      {props.messageDetails.prescriptionData && (
+        <MedicalPrescriptionIdentifiersComponent
+          prescriptionData={props.messageDetails.prescriptionData}
+        />
+      )}
 
-    {props.prescriptionData !== undefined && props.dueDate !== undefined && (
-      <MedicalPrescriptionDueDateBar
-        dueDate={props.dueDate}
-        messageDetails={props.messageDetails}
-        paymentExpirationInfo={props.paymentExpirationInfo}
-      />
-    )}
-  </>
-);
+      {props.messageDetails.prescriptionData === undefined &&
+        props.messageDetails.dueDate !== undefined && (
+          <DueDateBar
+            dueDate={props.messageDetails.dueDate}
+            expirationInfo={paymentExpirationInfo}
+            isPaid={props.hasPaidBadge}
+          />
+        )}
+
+      {props.messageDetails.prescriptionData !== undefined &&
+        props.messageDetails.dueDate !== undefined && (
+          <MedicalPrescriptionDueDateBar
+            dueDate={props.messageDetails.dueDate}
+            messageDetails={props.messageDetails}
+            paymentExpirationInfo={paymentExpirationInfo}
+          />
+        )}
```

Comment:
I guess these components are exclusive when `props.messageDetails.dueDate !== undefined`
I find this refactoring more readable 

```suggestion
      {props.messageDetails.dueDate !== undefined &&
        (props.messageDetails.prescriptionData === undefined ? (
          <DueDateBar
            dueDate={props.messageDetails.dueDate}
            expirationInfo={paymentExpirationInfo}
            isPaid={props.hasPaidBadge}
          />
        ) : (
          <MedicalPrescriptionDueDateBar
            dueDate={props.messageDetails.dueDate}
            messageDetails={props.messageDetails}
            paymentExpirationInfo={paymentExpirationInfo}
          />
        ))}
```
It's just refactoring. Feel free to ignore if you don't like it

Reply:
refactored in 78202a50ccd864b9c9f7b1845a4361ce190b6ba2